File: Python/Stadtrad.py
from Fahrrad import Fahrrad

class Stadtrad(Fahrrad):
	def __init__(self,fahrer='',farbe='',v=0,richtung=0,licht=False):
		Fahrrad.__init__(self,fahrer,farbe,v,richtung)
		self.setLicht(licht)

	def __str__(self):
		return ';'.join([Fahrrad.__str__(self),'Licht: '+str(self.licht)])

	def setLicht(self,licht):
		licht=bool(licht)
		self.licht=licht

if __name__=='__main__':
	r=Stadtrad('Thomas','schwarz',licht=False)
	r.lenken(10)
	r.beschleunigen(0.3,9.8)
	r.setLicht(True)
	print r

File: Java/src/biz/no_ip/druidamortis/Labs/Stadtrad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package biz.no_ip.druidamortis.Labs;

/**
 *
 * @author michael
 */
public class Stadtrad extends Fahrrad{
    private boolean licht;

    public void setLicht(boolean licht){
        this.licht = licht;
    }
    public boolean isLicht(){
        return this.licht;
    }
    
    
    
    
    public Stadtrad(String fahrer, String farbe, int richtung, double geschwindigkeit, boolean licht){
        super(fahrer, farbe, richtung, geschwindigkeit);
        this.licht = licht;
    }
    public Stadtrad(String fahrer, String farbe, boolean licht) {
        this(fahrer, farbe, 0, 0, licht);
        //oder super(fahrer, farbe); this.licht = licht;
    }
    public Stadtrad(){
        
    }
    
    @Override
    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append("Fahrer: ").append(this.getFahrer()).append(", Farbe: ").append(this.getFarbe()).append(", Richtung: ").append(this.getRichtung()).append(", km/h: ").append(this.getKmh()).append(", Licht: ");
        if(isLicht())
            sb.append("an");
        if(!isLicht())
            sb.append("aus");
        return sb.toString();
    }
    
    
    public static void main(String args[]) {
        Fahrrad r = new Stadtrad("Thomas","schwarz", false); r.lenken(10);
        r.beschleunigen(0.3,9.8); ((Stadtrad)r).setLicht(true);
        System.out.println(r);
    }
}
